### PR TITLE
helmified imagePullSecrets and resources

### PR DESCRIPTION
**Pull request by twhite0**
#### What type of PR is this?

- [x] Enhancement / New Feature
- [ ] Bug Fix
- [ ] Refactor
- [ ] Optimization
- [ ] Test
- [ ] Documentation Update

#### What this PR does / why we need it:
Working in my corporate environment surfaced two items that would be beneficial to allow tailoring within the helm chart.  
The first is providing the ability to supply an `imagePullSecret`; our internal image repository requires credentials to pull images.  
The second is the ability to override the operator and proxy resources.  
While deploying the operator in our environment we were receiving OOM/137's on the operator container.  
While I still have some work to do to understand why I was crashing using the defaults, having the ability to adjust the resources helped in the initial deployment.

#### Which issue(s) this PR fixes:

Fixes https://github.com/istio-ecosystem/sail-operator/issues/354

Related Issue/PR #

#### Special notes for your reviewer:
```text
Struggling with how to provide test cases for my changes?  

##### added pull secret to serviceaccount
$ helm template helm-default chart --set imagePullSecrets="{mypullsecret}" | grep -A 7 "app.kubernetes.io/name: serviceaccount"
    app.kubernetes.io/name: serviceaccount
    app.kubernetes.io/instance: sail-operator
    app.kubernetes.io/component: rbac
    app.kubernetes.io/managed-by: helm
  name: sail-operator
  namespace: default
imagePullSecrets:
  - name: mypullsecret


##### default no pull secret
$ helm template helm-default chart | grep -A 7 "app.kubernetes.io/name: serviceaccount"
    app.kubernetes.io/name: serviceaccount
    app.kubernetes.io/instance: sail-operator
    app.kubernetes.io/component: rbac
    app.kubernetes.io/managed-by: helm
  name: sail-operator
  namespace: default
---
# Source: sail-operator/templates/rbac/auth_proxy_client_clusterrole.yaml

##### added resources
$ helm template helm-default chart --set resources.limits.cpu=1000m --set resources.limits.memory=1000Mi --set resources.requests.cpu=500m --set resources.requests.memory=500Mi --set proxy.resources.limits.cpu=750m --set proxy.resources.limits.memory=750Mi --set proxy.resources.requests.cpu=250m --set proxy.resources.requests.memory=250Mi  | grep -B 5 -A 6 "     resources"
        name: kube-rbac-proxy
        ports:
        - containerPort: 8443
          name: https
          protocol: TCP
        resources:
          limits:
            cpu: 750m
            memory: 750Mi
          requests:
            cpu: 250m
            memory: 250Mi
--
          httpGet:
            path: /readyz
            port: 8081
          initialDelaySeconds: 5
          periodSeconds: 10
        resources:
          limits:
            cpu: 1000m
            memory: 1000Mi
          requests:
            cpu: 500m
            memory: 500Mi

##### default resources

$ helm template helm-default chart | grep -B 5 -A 6 "     resources"
        name: kube-rbac-proxy
        ports:
        - containerPort: 8443
          name: https
          protocol: TCP
        resources:
          limits:
            cpu: 500m
            memory: 128Mi
          requests:
            cpu: 5m
            memory: 64Mi
--
          httpGet:
            path: /readyz
            port: 8081
          initialDelaySeconds: 5
          periodSeconds: 10
        resources:
          limits:
            cpu: 500m
            memory: 512Mi
          requests:
            cpu: 10m
            memory: 64Mi

```

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation:
```text

```

#### Does this PR introduce a breaking change?
```text
No
```

#### Other information:
```text

```

### CODE REVIEW

**Comment by fjglira** [comment]
Hey @twhite0 , you will need to update this PR because you already merged on this [one](https://github.com/istio-ecosystem/sail-operator/pull/375/files) the capability to set custom resources

**Comment by twhite0** [comment]
Resources changes outlined previously were implemented in https://github.com/istio-ecosystem/sail-operator/pull/375

File: chart/templates/rbac/service_account.yaml [hunk 1]
```diff
@@ -8,3 +8,9 @@ metadata:
     app.kubernetes.io/managed-by: helm
   name: {{ .Values.serviceAccountName }}
   namespace: {{ .Release.Namespace }}
+  {{- if .Values.imagePullSecrets }}
+imagePullSecrets:
+  {{- range .Values.imagePullSecrets }}
+  - name: {{ . }}
+  {{- end }}
+  {{- end }}
```
Comment (dgn) [inline chart/templates/rbac/service_account.yaml:16]: nit: missing newline

**Review by dgn — COMMENTED** [review]
lgtm, just one small nit

File: chart/templates/rbac/service_account.yaml [hunk 2]
```diff
@@ -8,3 +8,10 @@ metadata:
...
+
```
Comment (dgn) [inline chart/templates/rbac/service_account.yaml:17]: strange. it still complains about missing newline. I guess it's not that big of a deal

**Review by dgn — APPROVED** [review]

**Comment by dgn** [comment]
This will just need a rebase

**Pull request merged**